compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/generators/CallAndReferenceGenerator.kt

```diff
@@ -500,7 +500,7 @@ class CallAndReferenceGenerator(
     }
 
     private fun IrMemberAccessExpression<*>.applyArgumentsWithReorderingIfNeeded(
-        argumentMapping: LinkedHashMap<FirExpression, FirValueParameter>,
```

Comment:
I guess the predictable iteration order matters. Very unlikely, but if arguments are really many, and if those keys are sorted, then we lost the argument order.

If this change is just to cover `FirPartiallyResolvedArgumentList` with unordered mapping, we can ignore that. I don't see the point of converting partially resolved FIR argument mapping to BE IR, because it's wrong anyway.

Reply:
The order _does_ matter. I changed this in https://github.com/JetBrains/kotlin/pull/3755. While `mutableMapOf()` does create a `LinkedHashMap`, we cannot rely on this implementation detail which we had been doing so. I changed the type to explicitly use `LinkedHashMap` to signify the order is important.

Reply:
OK, the only reason I changed the type is to avoid type cast in `FirPartiallyResolvedArgumentList`. But I can do that.